README.md

```diff
@@ -1,2 +1,331 @@
+
+Farcaster is a [sufficiently decentralized](https://www.varunsrinivasan.com/2022/01/11/sufficient-decentralization-for-social-networks) protocol that empowers developers to build novel social networks. We define a sufficiently decentralized network as one where **two users who want to communicate are always able to, even if the network wants to prevent it**. Users on such networks must have full control over their identity (usernames), data (messages) and social graph (relationships to others). If a third party controls any of these, they can prevent two users from communicating. Developers must also be free to build applications and have unrestricted access to the network and users must be free to switch between them. If there was only one app to connect to the network, it could prevent two users from communicating.  
+
+# 2. Concepts
+
+Farcaster achieves sufficient decentralization through a hybrid architecture that has on-chain and off-chain components. 
+
+User identities are kept on-chain because they are valuable assets, and we want to leverage the security, composability and strong consistency guarantees of the Ethereum blockchain. On-chain identities are controlled by an Ethereum address, which is a public private key pair that can also sign off-chain messages.  
+
+User data cannot be practically stored on the Ethereum blockchain or any other L1 because of the strong privacy and scalability requirements. Instead users store their data off-chain on a server under their control known as a Farcaster Hub. All data must be cryptographically signed by the user's identity address before being published to the Hub. 
+
+<!-- Diagram covering the major architectural concepts  -->
+
+## 2.1 Accounts
+
+An Farcaster account is similar to an account on pseudonymous social networks like Twitter or Reddit. An individual can operate several accounts as the same time, like a real-name account, a pseudonymous account and a company account.
+
+An account is created by minting a new account number from the Account contract on-chain. Each account number is like a primary key in a database, and points to a user-controlled address known as the `custody address`. Once an account is minted, the custody address can sign messages on behalf of the account. Accounts can be enriched by adding a profile picture, display name, biography and verified usernames like `alice.eth`, which are all done off-chain with signed messages. 
+
+## 2.2 Signed Messages
+
+Signed Messages are cryptographically signed messages from an account that are **tamper-proof** and **self-authenticating**. 
```

Comment:
Defining "message" with "message" is a little self-referential. Is there another term that can be used ?
"Signed data containers", "Signed data blobs", "Signed JSON objects", ... ?